cocos/rendering/custom/web-pipeline-types.ts

```diff
@@ -901,37 +901,48 @@ export class RenderDrawQueue {
 }
 
 export class RenderInstancingQueue {
-    /**
-     * @en A set of instanced buffer
-     * @zh Instance 合批缓存集合。
-     */
-    public queue = new Set<InstancedBuffer>();
+    passInstances: Map<Pass, number> = new Map<Pass, number>();
+    instanceBuffers: Array<InstancedBuffer> = new Array<InstancedBuffer>();
+
     empty (): boolean {
-        return this.queue.size === 0;
+        return this.passInstances.size === 0;
     }
 
     add (pass: Pass, subModel: SubModel, passID: number): void {
-        const instancedBuffer = pass.getInstancedBuffer();
+        const iter = this.passInstances.get(pass);
+        if (iter === undefined) {
+            const instanceBufferID = this.passInstances.size;
+            if (instanceBufferID >= this.instanceBuffers.length) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+                this.instanceBuffers.push(new InstancedBuffer(pass));
+            }
+            this.passInstances.set(pass, instanceBufferID);
+
+            const instanceBuffer = this.instanceBuffers[instanceBufferID];
+            instanceBuffer.pass = pass;
+            const instances = instanceBuffer.instances;
+        }
+
+        const instancedBuffer = this.instanceBuffers[this.passInstances.get(pass)!];
         instancedBuffer.merge(subModel, passID);
-        this.queue.add(instancedBuffer);
     }
 
     clear (): void {
-        const it = this.queue.values(); let res = it.next();
-        while (!res.done) {
-            res.value.clear();
-            res = it.next();
-        }
-        this.queue.clear();
+        this.passInstances.clear();
+        const instanceBuffers = this.instanceBuffers;
+        instanceBuffers.forEach((instance) => {
+            instance.clear();
+        });
```

Comment:
Should also clear instanceBuffers?

Reply:
I think gfx instance buffers are reused. They are not recreated every frame.

Reply:
The instanceBuffers here will be reused, so there's no need to clear them

Reply:
And the size of it is not changed?

Reply:
The reuse logic is here： ![image](https://github.com/user-attachments/assets/fea979c0-b10c-44ee-857d-5474aefc48e0)